test/csuite/random_directio/main.c

```diff
@@ -0,0 +1,1264 @@
+	bool populate_only, rand_th, rand_time, verify_only;
+
+	(void)testutil_set_progname(argv);
+
+	datasize = DEFAULT_DATA_SIZE;
+	nth = MIN_TH;
+	ncycles = DEFAULT_CYCLES;
+	rand_th = rand_time = true;
+	timeout = MIN_TIME;
+	interval = DEFAULT_INTERVAL;
+	flags = 0;
+	populate_only = verify_only = false;
+	working_dir = "WT_TEST.random-directio";
+	method = "none";
+	pid = 0;
+	WT_CLEAR(args);
+
+	if (!has_direct_io()) {
+		fprintf(stderr, "**** test_random_directio: this system does "
+		    "not support direct I/O.\n**** Skipping test.\n");
+		return (EXIT_SUCCESS);
+	}
+	for (i = 0, p = args; i < (uint32_t)argc; i++) {
+		testutil_check(__wt_snprintf_len_set(p,
+		    sizeof(args) - (size_t)(p - args), &size, " %s",
+		    argv[i]));
+		p += size;
+	}
+	while ((ch = __wt_getopt(progname, argc, argv,
+	    "d:h:i:m:n:pS:T:t:v")) != EOF)
+		switch (ch) {
+		case 'd':
+			datasize = (uint32_t)atoi(__wt_optarg);
+			if (datasize > LARGE_WRITE_SIZE ||
+			    datasize < MIN_DATA_SIZE) {
+				fprintf(stderr,
+				    "-d value is larger than maximum %"
+				    PRId32 "\n",
+				    LARGE_WRITE_SIZE);
+				return (EXIT_FAILURE);
+			}
+			break;
+		case 'h':
+			working_dir = __wt_optarg;
+			break;
+		case 'i':
+			interval = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'm':
+			method = __wt_optarg;
+			if (!WT_STREQ(method, "fsync") &&
+			    !WT_STREQ(method, "dsync") &&
+			    !WT_STREQ(method, "none")) {
+				fprintf(stderr,
+				    "-m option requires fsync|dsync|none\n");
+				return (EXIT_FAILURE);
+			}
+			break;
+		case 'n':
+			ncycles = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'p':
+			populate_only = true;
+			break;
+		case 'S':
+			p = __wt_optarg;
+			while ((arg = strtok_r(p, ",", &p)) != NULL) {
+				if (WT_STREQ(arg, "all"))
+					LF_SET(SCHEMA_ALL);
+				else if (WT_STREQ(arg, "create"))
+					LF_SET(SCHEMA_CREATE);
+				else if (WT_STREQ(arg, "create_check"))
+					LF_SET(SCHEMA_CREATE_CHECK);
+				else if (WT_STREQ(arg, "data_check"))
+					LF_SET(SCHEMA_DATA_CHECK);
+				else if (WT_STREQ(arg, "drop"))
+					LF_SET(SCHEMA_DROP);
+				else if (WT_STREQ(arg, "drop_check"))
+					LF_SET(SCHEMA_DROP_CHECK);
+				else if (WT_STREQ(arg, "rename"))
+					LF_SET(SCHEMA_RENAME);
+				else if (WT_STREQ(arg, "verbose"))
+					LF_SET(SCHEMA_VERBOSE);
+				else if (WT_STREQ(arg, "none"))
+					flags = 0;
+				else {
+					fprintf(stderr,
+					    "Unknown -S arg '%s'\n", arg);
+					usage();
+				}
+			}
+			break;
+		case 'T':
+			rand_th = false;
+			nth = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 't':
+			rand_time = false;
+			timeout = (uint32_t)atoi(__wt_optarg);
+			break;
+		case 'v':
+			verify_only = true;
+			break;
+		default:
+			usage();
+		}
+	argc -= __wt_optind;
+	if (argc != 0)
+		usage();
+
+	testutil_work_dir_from_path(home, sizeof(home), working_dir);
+	/*
+	 * If the user wants to verify they need to tell us how many threads
+	 * there were so we can find the old record files.
```

Comment:
This comment is similar to, or from `random_abort` is "old record files" the right phrasing here?